src/context/NotificationsContext/NotificationsContextProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { type ReactNode, useEffect, useMemo } from 'react';
+
+import { useCreateUserNotificationsDataMutation } from '~gql';
+
+import { useAppContext } from '../AppContext/AppContext.ts';
+
+import { NotificationsContext } from './NotificationsContext.ts';
+
+const NotificationsContextProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const { updateUser, user } = useAppContext();
+  const [createUserNotificationsData] =
+    useCreateUserNotificationsDataMutation();
+
+  useEffect(() => {
```

Comment:
I wonder if it's all right to perform this effect inside the AppContextProvider instead since that seems to handle user update operations as well. I see the value with decoupling the logic for notifications though but I wonder if it's worth it since there's technically nothing to consume from this Provider

But then again, if we do have something to consume in the future then totally worth keeping it the way it is

Reply:
@rumzledz I think it totally makes sense to keep this provider as the idea would be in the future to display the notifications in the user hub

Reply:
Oh right thanks @mmioana , in that case I shall use this as we can all forget about everything else I said 😂 
"But then again, if we do have something to consume in the future then totally worth keeping it the way it is"

Reply:
Yep, that's exactly it, this notifications provider is going grow in the future. The plan is for it to be an abstraction so we don't have magicbell code throughout the app, but instead have it only in one place. That way if we ever need to change from using magicbell to a different notifications provider (or even build one in-house) then we only have to change it in this context, not everywhere notifications are used.